pkg/config/config.go

```diff
@@ -150,12 +145,36 @@ func (gc *Config) ShouldSkipTLSForAuthority(candidateAuthority string) bool {
 }
 
 func (gc *Config) addTrustedCerts(certChain string) (err error) {
+	backupCertsFilePath := backupSystemCertsFile
+	systemCertsFilePath := systemCertsFile
+
+	if gc.BackupCaBundlePath != "" && gc.SystemCaBundlePath != "" {
+		backupCertsFilePath = gc.BackupCaBundlePath
+		systemCertsFilePath = gc.SystemCaBundlePath
+	}
+
+	src, err := os.OpenFile(backupCertsFilePath, os.O_RDONLY, os.ModeExclusive)
+	if err != nil {
+		return fmt.Errorf("Opening original certs file: %s", err)
+	}
+	defer src.Close()
+
+	dst, err := os.OpenFile(systemCertsFilePath, os.O_CREATE|os.O_RDWR|os.O_TRUNC, os.ModeExclusive)
+	if err != nil {
+		return fmt.Errorf("Opening certs file: %s", err)
+	}
+	defer dst.Close()
+
+	_, err = io.Copy(dst, src)
+	if err != nil {
+		return fmt.Errorf("Copying certs file: %s", err)
+	}
+
 	if certChain == "" {
 		return nil
 	}
 
-	var file *os.File
-	file, err = os.OpenFile(systemCertsFile, os.O_CREATE|os.O_APPEND|os.O_WRONLY, os.ModeAppend)
+	file, err := os.OpenFile(systemCertsFilePath, os.O_APPEND|os.O_WRONLY, os.ModeAppend)
```

Comment:
in general this seems racy to me in combination with above.

Reply:
(specifically, initial write or append are not atomic to readers of the file so they may actually see corrupt view of the file while we are modifying it)

Reply:
@neil-hickey let me know if you want to talk about this.